pkg/piecestore/rpc/server/api/api.go

```diff
@@ -0,0 +1,175 @@
+// Copyright (C) 2018 Storj Labs, Inc.
+// See LICENSE for copying information.
+
+package api
+
+import (
+	"bytes"
+	"fmt"
+	"io"
+	"log"
+	"os"
+
+	"golang.org/x/net/context"
+
+	"storj.io/storj/pkg/piecestore"
+	"storj.io/storj/pkg/piecestore/rpc/server/ttl"
+	pb "storj.io/storj/protos/piecestore"
+)
+
+// Server -- GRPC server meta data used in route calls
+type Server struct {
+	PieceStoreDir string
+	DB            *ttl.TTL
+}
+
+// Store -- Store incoming data using piecestore
+func (s *Server) Store(stream pb.PieceStoreRoutes_StoreServer) error {
+	log.Println("Storing data...")
+
+	total := int64(0)
+
+	// Receive initial meta data about what's being stored
+	piece, err := stream.Recv()
+	if err == io.EOF {
+		return err
+	} else if err != nil {
+		return err
+	}
+
+	err = s.DB.AddTTLToDB(piece.Id, piece.Ttl)
+	if err != nil {
+		return err
+	}
+
+	// Initialize file for storing data
+	storeFile, err := pstore.StoreWriter(piece.Id, piece.Size, piece.StoreOffset, s.PieceStoreDir)
+	if err != nil {
+		return err
+	}
+	defer storeFile.Close()
+
+	for {
+		storeMsg, err := stream.Recv()
+		if err == io.EOF {
+			break
+		} else if err != nil {
+			return err
+		}
+
+		// Write chunk received to disk
+		n, err := storeFile.Write(storeMsg.Content)
+		if err != nil {
+			return err
+		}
+
+		total += int64(n)
+	}
+
+	if total < piece.Size {
+		return fmt.Errorf("Recieved %v bytes of total %v bytes", total, piece.Size)
+	}
+
+	log.Println("Successfully stored data.")
+
+	return stream.SendAndClose(&pb.PieceStoreSummary{Message: "OK", TotalReceived: total})
```

Comment:
make the "OK"  a const so other packages can check that so if you want/need to change the message you can  do that without breaking users of this package.

Reply:
done